### PR TITLE
Preserve the active tab when saving settings

### LINKED ISSUE #18: Consider to set the active tab when saving
On the setting page, when saving, after the page reloads the previously active tab should be active. Currently instead, you're always returned to the first tab.

To reproduce:
- go for example on the "Clean Emails" tab
- press the "Save Changes" button
- after page reloads, I'd expect to be returned to the "Clean Emails" tab
- instead, the active tab is the first one

The current implementation misses the part to get the current hash value, store it somewhere, get it again after page reloads and use it to set the active tab. Seems it's a simple improvement, just needs to use the same implementation from Yoast SEO. On it.

### PR DESCRIPTION
**Pull request by afercia**
Uses the same implementation from Yoast SEO to preserve the active tab when saving settings and the admin page reloads.

To test:
- go in the settings page
- go in a tab other than the first one
- press "Save Changes"
- after the page reload, check the active tab is the one that was displayed before saving

Fixes #18

### CODE REVIEW

File: admin/assets/js/yoast-comment-hacks.js [hunk 1]
```diff
@@ -10,16 +11,49 @@ jQuery(document).ready(function() {
 		jQuery(this).addClass('nav-tab-active');
 	});
 
-	// init
-	var activeTab = window.location.hash.replace('#top#', '');
+	/**
+	 * Makes sure we store the action hash so we can return to the right hash.
+	 */
+	function wpseoSetTabHash() {
```
Comment (jdevalk) [inline admin/assets/js/yoast-comment-hacks.js]: can we rename this to `yoastCHSetTabHash()` ?

File: admin/assets/js/yoast-comment-hacks.js [hunk 2]
```diff
@@ -10,16 +11,49 @@ jQuery(document).ready(function() {
...
+			conf.attr( 'action', currentUrl + window.location.hash );
+		}
 	}
 
-	jQuery('#' + activeTab).addClass('active');
-	jQuery('#' + activeTab + '-tab').addClass('nav-tab-active');
+	/**
+	 * When the hash changes, get the base url from the action and then add the current hash.
+	 */
+	jQuery( window ).on( 'hashchange', wpseoSetTabHash );
```
Comment (jdevalk) [inline admin/assets/js/yoast-comment-hacks.js]: and then use `yoastCHSetTabHash` here too.

File: admin/assets/js/yoast-comment-hacks.js [hunk 3]
```diff
@@ -10,16 +11,49 @@ jQuery(document).ready(function() {
...
+		}
+
+		jQuery( '#' + activeTabId ).addClass( 'active' );
+		jQuery( '#' + activeTabId + '-tab' ).addClass( 'nav-tab-active' ).click();
+	}
 
-	jQuery('.nav-tab-active').click();
-});
+	// When the hash changes, get the base url from the form action and then add the current hash to the url.
+	wpseoSetTabHash();
```
Comment (jdevalk) [inline admin/assets/js/yoast-comment-hacks.js]: and use `yoastCHSetTabHash` here too.

File: admin/assets/js/yoast-comment-hacks.js [hunk 4]
```diff
@@ -10,16 +11,49 @@ jQuery(document).ready(function() {
 		jQuery(this).addClass('nav-tab-active');
 	});
 
-	// init
-	var activeTab = window.location.hash.replace('#top#', '');
+	/**
+	 * Makes sure we store the action hash so we can return to the right hash.
+	 */
+	function wpseoSetTabHash() {
+		var conf = jQuery( '#wpseo-conf' ),
```
Comment (jdevalk) [inline admin/assets/js/yoast-comment-hacks.js]: We should use `#yoast-ch-conf`

File: admin/views/config-page.php [hunk 5]
```diff
@@ -18,7 +18,7 @@
 			   href="#top#clean-emails"><?php _e( 'Clean emails', 'yoast-comment-hacks' ); ?></a>
 		</h2>
 
-		<form action="<?php echo esc_url( admin_url( 'options.php' ) ); ?>" method="post">
+		<form action="<?php echo esc_url( admin_url( 'options.php' ) ); ?>" id="wpseo-conf" method="post">
```
Comment (jdevalk) [inline admin/views/config-page.php]: We shouldn't use `wpseo-conf` but `yoast-ch-conf`

File: admin/assets/js/yoast-comment-hacks.js [hunk 6]
```diff
@@ -10,16 +11,49 @@ jQuery(document).ready(function() {
...
+	/**
+	 * Makes sure we store the action hash so we can return to the right hash.
+	 */
+	function yoastCHSetTabHash() {
+		var conf = jQuery( '#yoast-ch-conf' ),
+			currentUrl;
 
-	// default to first tab
-	if (activeTab === '' || activeTab === '#_=_') {
-		activeTab = jQuery('.wpseotab').attr('id');
```
Comment (jdevalk) [inline admin/assets/js/yoast-comment-hacks.js]: One more here, sorry, missed this last time :D

File: admin/assets/js/yoast-comment-hacks.js [hunk 7]
```diff
@@ -10,16 +11,49 @@ jQuery(document).ready(function() {
...
+		 * `#wpbody-content` and other plugins may do that as well. Also, facebook
+		 * adds a `#_=_` see wordpress-seo PR 506. In these cases and when it's
+		 * empty, default to the first tab.
+		 */
+		if ( '' === activeTabId || '#' === activeTabId.charAt( 0 ) ) {
+			/*
+			 * Reminder: jQuery attr() gets the attribute value for only the first
+			 * element in the matched set so this will always be the first tab id.
+			 */
+			activeTabId = jQuery( '.wpseotab' ).attr( 'id' );
```
Comment (jdevalk) [inline admin/assets/js/yoast-comment-hacks.js]: And here too.

**Comment by afercia** [comment]
Yeah sorry, too much copy-pasta 🍝

**Comment by jdevalk** [comment]
Merged manually as I've been doing a lot of cleanup.

**Pull request merged**